### PR TITLE
add pruning ticker for stale volumes on disk, unneeded secret/configmap informers

### PR DESCRIPTION
**Pull request by gabemontero**
should be the last of those meets min, miscellaneous clean up items I've been alluding to in scrum / office hours

/assign @coreydaley

### CODE REVIEW

File: pkg/hostpath/hostpath.go
Comment (coreydaley) [inline pkg/hostpath/hostpath.go:873]: Why not just attempt to delete the file(s) right after you check if the pod exists or not, instead of iterating over a separate slice when you are done?

**Review by coreydaley — CHANGES_REQUESTED** [review]

File: pkg/hostpath/hostpath.go
Comment (gabemontero) [reply pkg/hostpath/hostpath.go:873]: I was paranoid about messing up the walk function if we deleted items from the list it was iterating over while we were in the middle of iterating (old Java pain points ;-)) 

Trying some internet searches now to confirm or deny that paranoia ....

Comment (gabemontero) [reply pkg/hostpath/hostpath.go:873]: found nothing conclusive - given this runs only once per 10 minutes, I'm leaning toward caution and siding toward paranoia vs. efficiency

I'm putting a comment to explain rationale

that work for you?

Comment (coreydaley) [reply pkg/hostpath/hostpath.go:873]: Yep, that works for me.

Comment (gabemontero) [reply pkg/hostpath/hostpath.go:874]: cool thanks

**Comment by gabemontero** [comment]
updates pushed @coredaley

**Comment by gabemontero** [comment]
terraform fail with slow e2e

/test e2e-aws-csi-driver-slow

**Comment by coreydaley** [comment]
/retest
 Get "https://api.ci-op-g86ktcx7-06790.origin-ci-int-aws.dev.rhcloud.com:6443/apis/config.openshift.io/v1/clusterversions/version": dial tcp 54.197.50.168:6443: i/o timeout

**Comment by gabemontero** [comment]
this time cluster install failures 

/retest

**Comment by gabemontero** [comment]
/hold

yeah I'm seeing intermittent failures in the base e2e locally now as well @coreydaley

Think we are beyond CI / AWS hiccups and MAYBE my change here is doing something wrong (I've seen passing jobs in other PRs today)?  Or we are hitting an intermittent problem independent of this change :-/ .....let's hope it is not the latter

investigating ....

**Comment by gabemontero** [comment]
> /hold
> 
> yeah I'm seeing intermittent failures in the base e2e locally now as well @coreydaley
> 
> Think we are beyond CI / AWS hiccups and MAYBE my change here is doing something wrong (I've seen passing jobs in other PRs today)? Or we are hitting an intermittent problem independent of this change :-/ .....let's hope it is not the latter
> 
> investigating ....

maybe it is slow image pulls for registry.access.redhat.com:

```
                "containerStatuses": [
                    {
                        "name": "my-frontend",
                        "state": {
                            "waiting": {
                                "reason": "ContainerCreating"
                            }
                        },
                        "lastState": {},
                        "ready": false,
                        "restartCount": 0,
                        "image": "registry.access.redhat.com/ubi8/ubi",
                        "imageID": "",
                        "started": false
                    }
                ],
```

Going to add some more failure debug, maybe increase the wait time for a pod to get ready.

**Comment by gabemontero** [comment]
Yeah it took 35 seconds to pull the ubi8 image in my local repro, and I never see the successfully pulled event, only the pulling event, in https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_csi-driver-shared-resource/82/pull-ci-openshift-csi-driver-shared-resource-master-e2e-aws-csi-driver/1468946435180335104/build-log.txt

I'm going to switch to registry.redhat.io for the image (that is suppose to be our "primary" registry with more beef to it) and see about adding a monitor for the image pull events prior to tracking the pod going to running @coreydaley

**Comment by gabemontero** [comment]
ok @coreydaley added a new commit that switches the test pod image to registry.redhat.io and tracking image pulling before pod state

worked better for me locally

let's see how it does in CI

PTAL

**Comment by gabemontero** [comment]
/assign @prietyc123 

for qe-approved label

a suggesting for adding a test scenario for this @prietyc123 ... note, I think this is too difficult to automate, and you'll have to do maually
1) create a pod that uses a share 
2) determine which node it is on
3) use `oc rsh -c hostpath ...` to the shared resource driver pod on the same node as the test pod, grep the volume files under `/csi-volumes-map` and find the volume file that references your pod
4) then delete the test pod with `--force` ... I believe this should do an immediate delete where the kubelet does not ask the driver to unmount the volume
5) wait 10 minutes for the new pruner here to fire
6) look again in your oc rsh session and confirm that volume file for your test pod has been removed.

**Comment by gabemontero** [comment]
/assign @rolfedh 

for docs approved label

the PR addresses the type of functional details that are too detailed, implementation specfic, to mention in the docs

so no docs impact for this one

**Comment by gabemontero** [comment]
terraform failure no-refresh

/test e2e-aws-csi-driver-no-refreshresource

**Comment by prietyc123** [comment]
> /assign @prietyc123
> 
> for qe-approved label
> 
> a suggesting for adding a test scenario for this @prietyc123 ... note, I think this is too difficult to automate, and you'll have to do maually
> 
> 1. create a pod that uses a share
> 2. determine which node it is on
> 3. use `oc rsh -c hostpath ...` to the shared resource driver pod on the same node as the test pod, grep the volume files under `/csi-volumes-map` and find the volume file that references your pod
> 4. then delete the test pod with `--force` ... I believe this should do an immediate delete where the kubelet does not ask the driver to unmount the volume
> 5. wait 10 minutes for the new pruner here to fire
> 6. look again in your oc rsh session and confirm that volume file for your test pod has been removed.

@gabemontero I have reviewed the scenario manually on this pr and it works as expected.
Before deleting test pods, I had the vol dir available inside the node
```
# ls -a /csi-volumes-map
.  ..  csi-c454bb43b8a6480e93816a023b193d9c61ce13f49450de3722c859fd06cad4a0
```
After doing force test pod deletion
```
# ls -a /csi-volumes-map
.  ..
```
I will also add this scenario, while covering e2e feature file in #62 

/label qe-approved

**Comment by gabemontero** [comment]
> > /assign @prietyc123
> > for qe-approved label
> > a suggesting for adding a test scenario for this @prietyc123 ... note, I think this is too difficult to automate, and you'll have to do maually
> > 
> > 1. create a pod that uses a share
> > 2. determine which node it is on
> > 3. use `oc rsh -c hostpath ...` to the shared resource driver pod on the same node as the test pod, grep the volume files under `/csi-volumes-map` and find the volume file that references your pod
> > 4. then delete the test pod with `--force` ... I believe this should do an immediate delete where the kubelet does not ask the driver to unmount the volume
> > 5. wait 10 minutes for the new pruner here to fire
> > 6. look again in your oc rsh session and confirm that volume file for your test pod has been removed.
> 
> @gabemontero I have reviewed the scenario manually on this pr and it works as expected. Before deleting test pods, I had the vol dir available inside the node
> 
> ```
> # ls -a /csi-volumes-map
> .  ..  csi-c454bb43b8a6480e93816a023b193d9c61ce13f49450de3722c859fd06cad4a0
> ```
> 
> After doing force test pod deletion
> 
> ```
> # ls -a /csi-volumes-map
> .  ..
> ```
> 
> I will also add this scenario, while covering e2e feature file in #62
> 
> /label qe-approved

sounds good @prietyc123 thanks

**Comment by gabemontero** [comment]
bump @coreydaley just need any final review comments from you or tag for merge - thanks

**Pull request merged**
